[GSSI] Improve section on using modules (#586)
hpcleuven/VscDocumentation: source/compute/software/using_software/cluster_modules.rst

Cluster modules
===============

Background
~~~~~~~~~~

> It is common for HPC cluster to be inhomogeneous, meaning there are

Reviewer: ```suggestion
It is common for HPC clusters to be inhomogeneous, meaning there are
```

Reviewer 2: I see your typo fix, and I'll raise you a more commonly used word:
```suggestion
It is common for HPC clusters to be heterogeneous, meaning there are
```

with the ``module avail`` command. The different software stacks mentioned
earlier are located in different directories (see the
:ref:`next section <manually_modifying_modulepath>` for more details), so in
order to make sure you are loading modules from the appropriate software stack,
the ``$MODULEPATH`` variable needs to contain the appropriate paths for the
node where you want to use a module.

Because working with the different directories containing different software
stacks is cumbersome, we advise users to rely on a cluster module to set
the ``$MODULEPATH`` variable on VSC sites with support for cluster modules.
The cluster module can be handled identically as other modules, but instead
of making executables or libraries available, its only purpose is to set up
your environment to make the correct modules available. Cluster modules are
always available and you can see which versions can be loaded by executing
``module avail cluster/``. The anatomy of the name of a cluster module is

::

   cluster/<cluster_name>/<partition_name>

or

::

   cluster/<cluster_name>

A few examples are ``cluster/wice/batch``, ``cluster/wice/gpu_h100``,

> ``cluster/gallade`` and ``cluster/dodrio/gpu_rome_a100_rhel9``.

Reviewer 2: To future-proof this a bit better, let's use one of the latest HPC-UGent Tier-2 clusters as example:
```suggestion
``cluster/shinx`` and ``cluster/dodrio/gpu_rome_a100_rhel9``.
```

      environment and on the login nodes. In other cases (for example when you
      ssh directly into a node), you will need to first load a cluster module
      yourself in order to make the correct modules available.

   .. tab-item:: UGent
      :sync: ug

      Note that the ``cluster`` module will always remain
      loaded, since it defines some important environment variables that point
      to the location of centrally installed software/modules, and others that
      are required for submitting jobs and interfacing with the cluster
      resource manager ( ``qsub``, ``qstat``, ...).

.. warning::

   If your jobscript contains the command ``module --force purge``, the
   cluster module will be unloaded and your ``$MODULEPATH`` will not contain
   the directory with the appropriate software stack. It will be necessary to
   load the correct cluster module or set your ``$MODULEPATH`` in another way.
   This is why we advise to not use ``module --force purge`` in your jobs,
   unless you are well aware of the consequences. Note that it is ok to
   execute ``module purge``, since the cluster modules are
   :ref:`sticky <module_purge>`.

> 

Reviewer 2: This subsection is quite long, this seems like a good place to inject another subsection

```suggestion

.. _searching_for_software:

Searching for software
~~~~~~~~~~~~~~~~~~~~~~

```

A common scenario is that you want to search through the installed modules for
a software package you need, while you are on a login node. There are two ways
this can be done. The software package that is used as an example is called
``CP2K``.

The first option is to load the cluster module corresponding to the node where
you eventually want to use a certain software package.

.. tab-set::
   :sync-group: vsc-sites

   .. tab-item:: KU Leuven/UHasselt
      :sync: kuluh

      If you are planning to run jobs on the wICE batch partition, the commmand is
      ``module load cluster/wice/batch``. Note that the previously loaded cluster
      module will be automatically unloaded: at most one cluster module can be
      loaded at a time.

   .. tab-item:: UGent
      :sync: ug

      If you are planning to run jobs on the donphan cluster, the command is
      ``module swap cluster/donphan``. The ``swap`` command makes sure the
      previously loaded cluster module is unloaded.

Now you can search for modules containing ``CP2K`` by executing (the search is

> not case sensitive):

Reviewer 2: ```suggestion
done *case insensitive*):
```

.. code-block:: shell

>    $ module avail CP2K

Reviewer 2: bring home the case insensitive point:
```suggestion
   $ module avail cp2k
